src/lib/storage/storage_manager.cpp

```diff
@@ -10,38 +10,41 @@
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager sm;
+  return sm;
 }
 
-void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
-}
+void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) { tables[name] = table; }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
+  if (tables.at(name) != NULL) {
```

Comment:
```suggestion
  if (tables.contains(name)) {
```
Theoretically, a situation may occur, where a `NULL` pointer was added to the `tables` (currently, there is no check for that in the `add_table` method). So, I think, the check with `contains` could be more robust.

Reply:
`nullptr` instead of `NULL`. Also, Peter is correct, `contains()` is much nicer here.